library/ssl_misc.h

```diff
@@ -2178,18 +2178,18 @@ static inline int mbedtls_ssl_tls13_sig_alg_is_supported(
     switch( sig_alg )
     {
 #if defined(MBEDTLS_PKCS1_V15)
-#if defined(MBEDTLS_SHA256_C)
+#if defined(MBEDTLS_HAS_ALG_SHA_256_VIA_MD_OR_PSA_BASED_ON_USE_PSA)
```

Comment:
Good catch! Totally missed that when writing up the issue's description. Should we move this function to a new header `library/ssl_tls13_generic.h` though, so that in the future `grep <stuff> library/ssl_tls13*` works as expected? Cc @ronald-cron-arm 

Also, should this function really be `static inline`? We tend to `static inline` too many things, which is not good for code size.

(Both completely pre-existing, so feel free to push back if you feel the scope of this PR is expanding too much.)

Reply:
I can move this function to `ssl_tls13_generic.h`, but on the other hand I see that there is more TLS 1.3 functions in this header: `e.g. mbedtls_ssl_tls13_sig_alg_for_cert_verify_is_supported(), mbedtls_ssl_sig_alg_is_received()`.

Also it is used in this header, so we would need to include `library/ssl_tls13_generic.h`.

I'll wait for @ronald-cron-arm input.

Reply:
Ok, if there are many functions in the same case, it's probably out of scope for this PR, when.